### ISSUE TITLE
[Footer] - Required links for government sites?

### ISSUE THREAD

**Issue by humancompanion** [uswds/uswds#998]
Hey there - We've been working on applying the standards for the footer over at USAJOBS and had some questions. 

Version 1 is the version we are most likely going with:
http://usajobs.github.io/design-system/footer/
1. I've been told that I must include the links mentioned in this article:
   http://www.digitalgov.gov/resources/required-web-content-and-links/

It is a bit unclear to me if all of these really apply to USAJOBS since it is not a site representing an agency. I suspect that is why the current site does not have the "Budget and Performance" link. However, I'm wondering how you are avoiding including these links. Are these requirements out of date? 
1. What is the collective opinion on sub-sections in the footer? We've deliberately gone with a fat footer and are exposing a good deal of our navigation there. The code didn't make it particularly easy to add sub-sections (for example, instances of .usa-footer-big ul li which applies to every list item in the footer maybe could have been .usa-footer-big ul > li to impact just the top level). So are fat footers frowned upon?
2. I noted issue #706 and migrated our design in that direction. Given that we hope to release global nav updates this quarter and won't be able to revisit those components for some time I wanted to be ahead of the curve instead of behind it. Party foul?

Thanks for your guidance.

**Comment by maya** [comment]
Hi @humancompanion, those are great questions about footers! Looping in @carodew who may have some insight around this.

**Comment by carodew** [comment]
@humancompanion It would be great if we could talk to the developer/developers involved in working with the subsections in the footer so we could learn more about how it was difficult.

As for the required links – at the time of the MVP launch, we hadn't had the resources to do enough usability testing or user research to determine how those links would best fit into the footer design we're offering here, so we left it pretty generic for the time being. As user needs dictate, we may look into adding more detailed guidance on this issue in future iterations of the footer.  

We'd be very interested in learning what you discover from your own usability tests, so please reach out whenever you have something to share. Not entirely sure what links will be required for your site. 

We're not against fat footers, but it depends on how well it works for your specific users and contexts, so it's hard to say whether or not that's the right solution for USAJOBS.

**Comment by humancompanion** [comment]
@carodew - Sorry for the long delay in my, now rather long, response. We ended up moving away from sub-sections and a fat footer. Thus the quibbles I had with the subsection code are moot at this point. 

We got to a usability test last week that included the footer. It was a tiny test with 7 folks and it wasn't all focused on the footer but, we did have a key takeaway: No one found the tiny, required links we inserted at the bottom (as we expected). We asked folks to find more information on the personal data we collect about them in hopes they would find the privacy policy link and no one could. Thus we're going to include some, or all, of the topics that those links point to as full articles in our help so that they can found with search. However, given that those links are required by law or policy I'm going to include them. I think it would be prudent for the Standards to mention those links. I certainly was not aware that some of them were required by law. 

I coded the component so here are my quibbles:

* As I mentioned in the initial comment the problem I immediately ran into when adding sub-sections was that the CSS didn't expect them and thus declarations like <code>.usa-footer-big ul li</code> applied to every list item in the footer including sub-section items. Thus those styles had to be defeated with a more specific override. It could be more defensively coded with <code>.usa-footer-big ul > li</code> so that the styles only applied one level down.

In general the Standards do a good job to avoid styling bare elements and instead add a class to the element and apply the style to that class. This allows the markup to change if necessary without having to update the CSS. The footer, for whatever reason, doesn't seem to follow that methodology quite as closely. 

But more importantly, given that I've since ripped out the sub-sections, it would be great if the styles and behavior within a component could be separated. For example, the footer is essentially an accordion and should use the same JS to behave like an accordion. But, currently it has to have separate JS because the accordion behaviors are attached to the class .usa-accordion:

<code>
$('[class^=usa-accordion]').each(function() {
    accordion($(this));
  });
</code>

But it isn't a simple matter of adding the .usa-accordion class to the footer because then you'd get the styles of the accordion as well and the footer wants to use a different drawer handle, etc. Ideally, the accordion functionality would be attached to a data attribute. Then that data attribute could be added to any component that had the right aria attributes and thus it would behave like an accordion. 

This article captures what I'm suggesting:
https://medium.com/@cannikin/introducing-event-focused-js-and-css-efjac-9b8a9dd7bfe3#.lg7fu7tl6

**Comment by carodew** [comment]
Thanks for the feedback, and for sharing the results of the research you did! 

We're going to be revisiting the footer design fairly soon and will take this feedback into account. 

I'm curious – do you think that most users are interested in finding information about the Privacy Act, FOIA, or other required links? My hypothesis is that there is not much desire for that content from the vast majority of users, and it probably makes sense to include them on the page in a way that does not impede the main user goals. At any rate, I do think there is a place for the Standards to address the required links in some fashion.

**Comment by humancompanion** [comment]
@carodew - The question we asked in hopes that test participants would find the Privacy Policy link was "Where might you click if you wanted to find out more about what data USAJOBS collects about you?" For what it is worth, I did ask a follow up question to the test participants who tried for a significant amount of time to find the Privacy Policy. The question was along the lines of "Is the data USAJOBS collects in relation to your account something that is important to you?" It was an ad-lib, not in our test script, and in retrospect it was not well phrased. 

One participant, a student, indicated it was important to her. However, she indicated that failing to find a link she would resort to searching the help center for "privacy". Thus in the end she would I suspect, given our decision to include that content in the search results for the help center, find the relevant content. 

In short, including those privacy and policy related links in "fine print" at the bottom of the footer complies with the laws and policies governing our site but doesn't distract from key tasks. As long as that information is discoverable in another way (i.e. search) I feel the user's needs are met.

**Comment by maya** [comment]
This is not on the current product roadmap, closing for now.
